PrivateMaps/FaireWeatherTweakEx.py

```diff
@@ -5161,42 +5170,156 @@ def afterGeneration():
+    numOfIterations = 3
+
+    def isPlotOutOfBounds(x, y):
+        if x < 0 or x >= mc.width or y < 0 or y >= mc.height:
+            return True
+        else:
+            return False
+
+    def calculateIceChanceFromAdjacent(x, y):
+        # More chance for Ice if adjacent tiles are Ice, Permafrost or map bounds
+        totalChance = initialChance
+        for direction in range(1, 5):
+            xx, yy = GetXYFromDirection(x, y, direction)
+            if not isPlotOutOfBounds(xx, yy):
+                if (xx, yy) in icePlots:
+                    totalChance = addProbability(totalChance, chanceIncreaseAdjacent)
+        for direction in range(5, 9):
+            xx, yy = GetXYFromDirection(x, y, direction)
+            if not isPlotOutOfBounds(xx, yy):
+                if (xx, yy) in icePlots:
+                    totalChance = addProbability(totalChance, chanceIncreaseDiagonal)
+        return totalChance
+
+    # used only for command-line tests
+    def printIceMap():
+        lineString = "Ice Cap Map:"
+        print lineString
+        # South Pole
+        for y in range(mc.iceCapMaxHeightSouth + 1):
+            lineString = ""
             for x in range(mc.width):
-                plot = mmap.plot(x,y)
-                if plot != 0 and plot.isWater() == True and PRand.random() < iceChance:
-                    plot.setFeatureType(featureIce,0)
-            iceChance *= .66
-    iceChance = 1.0
-    #for y in range(mc.height - 1,mc.height - 5,-1):
-    if abs(mc.topLattitude) == 90 and (mc.topLattitude != mc.bottomLattitude or mc.topLattitude > 0):
-        for y in range(mc.height - 1,mc.height - 5,-1):
+                if (x, y) in icePlots:
+                    if icePlots[(x, y)] == "FEATURE_ICE":
+                        lineString += 'v'
+                else:
+                    lineString += '.'
+            print lineString
+
+        lineString = ""
+        for i in range(mc.width):
+            lineString += '0'
+        print lineString
+
+        # North Pole
+        for y in range(mc.height - mc.iceCapMaxHeightNorth - 1, mc.height, 1):
+            lineString = ""
             for x in range(mc.width):
-                plot = mmap.plot(x,y)
-                if plot != 0 and plot.isWater() == True and PRand.random() < iceChance:
-                    plot.setFeatureType(featureIce,0)
-            iceChance *= .66 
-    
-    # TAC - Map scripts - koma13 - END
+                if (x, y) in icePlots:
+                    if icePlots[(x, y)] == "FEATURE_ICE":
+                        lineString += 'v'
+                else:
+                    lineString += '.'
+            print lineString
+        lineString = " "
+        print lineString
+        return
+
+    # In this dictionary we store Ice plots which should be generated on water and Permafrost tiles on land
+    # (both cause probability increase for neighbouring plots to become ice)
+    icePlots = {}
+    gc = CyGlobalContext()
+    mmap = gc.getMap()
```

Comment:
``gc``,``mmap`` and stuff like this are defined at a global scope within the map script. I now this is not a good coding practice from the people that came before us, ~~but reassinging them here might not be the best move.~~

I note the previous version of ``createIce`` did that as well. So, yeah, let's live with it.

Reply:
Agree, I tried to avoid wherever possible.
Though:
- I'm just not aware how to do it efficiently in current architecture
- It's not a narrow place when we speak about performance.

Reply:
>     * I'm just not aware how to do it efficiently in current architecture
>     * It's not a narrow place when we speak about performance.

Agreed. Improved code design and performance is indeed worthy topics, but comments to a pull request is not the place.